### ISSUE TITLE
/home/myself/images used instead of /imageroot

### ISSUE THREAD

**Issue by eduarddrenth** [UCLALibrary/docker-cantaloupe#56]
I have to do:

```
docker run -d -p 8182:8182 --name xxx -v /somedir:/home/myself/images cantaloupe
```
instead of:
```
docker run -d -p 8182:8182 --name xxx -v /somedir:/imageroot cantaloupe
```
In the container I see:
```
cantaloupe@b2cf3ced2acf:~$ cat /etc/cantaloupe.properties|grep myself
FilesystemSource.BasicLookupStrategy.path_prefix = /home/myself/images/
cantaloupe@b2cf3ced2acf:~$ 
```
In the source tree I see:
```
~/docker-cantaloupe$ grep myself configs/*
configs/cantaloupe.properties.default-4.0.2:CANTALOUPE_ENDPOINT_PUBLIC_AUTH_BASIC_USERNAME = myself
configs/cantaloupe.properties.default-4.0.3:CANTALOUPE_ENDPOINT_PUBLIC_AUTH_BASIC_USERNAME = myself
configs/cantaloupe.properties.default-4.1:CANTALOUPE_ENDPOINT_PUBLIC_AUTH_BASIC_USERNAME = myself
configs/cantaloupe.properties.default-4.1.1:CANTALOUPE_FILESYSTEMSOURCE_BASICLOOKUPSTRATEGY_PATH_PREFIX = /home/myself/images/
configs/cantaloupe.properties.default-4.1.2:CANTALOUPE_FILESYSTEMSOURCE_BASICLOOKUPSTRATEGY_PATH_PREFIX = /home/myself/images/
configs/cantaloupe.properties.default-dev:CANTALOUPE_FILESYSTEMSOURCE_BASICLOOKUPSTRATEGY_PATH_PREFIX = /home/myself/images/
```
Perhaps I'll do another pr to standardize on user cantaloupe, also in the docs.

**Comment by ksclarke** [comment]
Yes, I've been meaning to clean up the README example with better volume mounts. I'd be inclined to not put the mount in the container inside a home dir though. The Cantaloupe example is /imageroot but maybe it would make more sense in the /mnt structure?

It does looks like something wonky is going on with the configs. I'll take a look but if you get to it before me, PRs are certainly welcome! Thanks :-)

**Comment by eduarddrenth** [comment]
Most important for me would be correct docs and getting rid of myself :-).
Mounting in the home dir of cantaloupe or under /mnt or in /imageroot I care about less. Perhaps /mnt/imageroot is a good option, it's where mounts usualy go in x systems.

**Comment by ksclarke** [comment]
Maybe I misunderstood... is there something not working about using /imageroot for you and you _have_ to put it under your user under home? As it is, /imageroot should work and be mapped to whatever you pass as the first/local mount point:

https://github.com/UCLALibrary/docker-cantaloupe/blob/master/Dockerfile#L99

**Comment by ksclarke** [comment]
Oh, is it because you want your container user to have write privileges to that directory where the images are? Right now, the internal 'cantaloupe' user doesn't, but we could change that so that the internal user can write to that dir too.

**Comment by ksclarke** [comment]
I was thrown a bit because I didn't realize we had `/home/myself/images` in the configs (so thought that was something you were adding). I've created a PR for the default configs to make them use `/imageroot/` now too (we haven't paid much attention to them since we're just overwriting values). Hope that fixes it. If you did want write privs in that `/imagesroot/` dir, too, let me know.

https://github.com/UCLALibrary/docker-cantaloupe/pull/57

**Comment by eduarddrenth** [comment]
> Maybe I misunderstood... is there something not working about using /imageroot for you and you _have_ to put it under your user under home? As it is, /imageroot should work and be mapped to whatever you pass as the first/local mount point:
> 
> https://github.com/UCLALibrary/docker-cantaloupe/blob/master/Dockerfile#L99

As I said in the bug this:

```
docker run -d -p 8182:8182 --name xxx -v /somedir:/imageroot cantaloupe
```
isn't working because cantaloupe looks in /home/myself/images.

**Comment by eduarddrenth** [comment]
fixed

> Oh, is it because you want your container user to have write privileges to that directory where the images are? Right now, the internal 'cantaloupe' user doesn't, but we could change that so that the internal user can write to that dir too.
> 
> _Edit:_ Oh, and I see the configs could have a saner/better
> default: https://github.com/UCLALibrary/docker-cantaloupe/blob/master/configs/cantaloupe.properties.default-4.1.2#L139

for me, read access is ok

**Comment by eduarddrenth** [comment]
Fixed by #57

**Comment by ksclarke** [comment]
Yes, sorry I misunderstood the initial bug report. Glad the PR fixed it for you.
